tools/update-npm.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -o pipefail -o errexit
+
+# This script will detect if we're running with an npm that
+# does not support CI, and will upgrade it if so
+MINIMUM_VERSION=5.7.0
+CURRENT_VERSION=$(npm -v)
+
+echo "Ensuring npm version is >= ${MINIMUM_VERSION}"
+
+# Sort the versions using `sort`s version sort, and take the
+# first line
+NEWEST_VERSION="$(echo -e "${CURRENT_VERSION}\n${MINIMUM_VERSION}" | sort -V -r | head -n1)"
+
+# If the head is the same as the current, we are already >=
+# the minimum version
+if [ "${CURRENT_VERSION}" = "${NEWEST_VERSION}" ]; then
+	echo 'Npm already supports npm ci, not upgrading';
```

Comment:
I wonder if we can just run `npm ci --help` and see if it succeeds, so we don't need to care about versions management, but just if feature is present.

Reply:
That's a very interesting thought, feature-driven testing is more robust indeed.

Reply:
@roman-mazur great idea

```bash-session
# nvm use 6; npm ci --help &>/dev/null || echo "need to update npm"
Now using node v6.17.1 (npm v3.10.10)
need to update npm
# nvm use 10; npm ci --help &>/dev/null || echo "need to update npm"
Now using node v10.16.0 (npm v6.9.0)
```

Reply:
Yeah, but I also wonder if we should update specific base images instead.

Reply:
Brilliant idea. As for the updating base images, I've always planned to create specific ones, but never gotten round to it. As it is, I'd like to have this working first, and move onto that.

Reply:
Fair enough, let's proceed in this way.